database/database.go

```diff
@@ -0,0 +1,232 @@
+package database
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	obj "github.com/jcgallegdup/Concurrent-Document-Editor/objects"
+	"github.com/mxk/go-sqlite/sqlite3"
+)
+
+// The Database struct implements the Database interface
+type Database struct {
+	Path string
+}
+
+const createDBFile = "createDb.sql"
+
+// CreateEmptyDb creates an empty database and returns the struct
+func CreateEmptyDb(name string) Database {
+	os.Remove(name)
+
+	creates, err := ioutil.ReadFile(createDBFile)
+	createString := string(creates)
+	if err != nil {
+		panic(err)
+	}
+
+	conn, err := sqlite3.Open(name)
+	defer conn.Close()
+	if err != nil {
+		panic(err)
+	}
```

Comment:
I definitely thought about this, but I couldn't think of a clean way of doing it. As you said the `defer` statement should always be in this outer function so that line is going to have to stay. The more problematic thing is that in the event of an error each function returns a different thing (e.g `CreateUser` returns -1 as the `userID`, whereas `GetFileContent` returns a dummy file).

Reply:
If an error occurs, I think we don't return anything anyway since we call `panic()`; or are you referring to a different error?